### PR TITLE
added error handling for get_hip_id_by_health_id

### PR DESCRIPTION
**Pull request by khavinshankar**
## Proposed Changes

- added error handling for get_hip_id_by_health_id

### Associated Issue

Errors are out when hf_id is not present for a facility. This interferes with discharging a patient when he has an ABHA number but the facility doesn't have hf_id.

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/abdm/utils/api_call.py [hunk 1]
```diff
@@ -349,11 +349,16 @@ def __init__(self):
...
+        try:
+            return (
+                AbhaNumber.objects.filter(
+                    Q(abha_number=health_id) | Q(health_id=health_id)
+                )
+                .first()
+                .patientregistration.facility.healthfacility.hf_id
+            )
+        except Exception:
+            return None
```
Comment (rithviknishad) [inline care/abdm/utils/api_call.py:361]: use `.exists` to check if record exists

**Review by Ashesh3 — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

File: care/abdm/api/viewsets/healthid.py [hunk 2]
```diff
@@ -351,18 +351,24 @@ def link_via_qr(self, request):
...
+                        "name": data["name"],
+                        "gender": data["gender"],
+                        "dateOfBirth": str(datetime.strptime(data["dob"], "%d-%m-%Y"))[
+                            0:10
+                        ],
+                    }
+                )
+            except Exception as e:
+                return Response(
+                    {"message": "Failed to fetch modes", "reason": str(e)},
```
Comment (sainak) [inline care/abdm/api/viewsets/healthid.py:367]:
```suggestion
                    {"detail": "Failed to fetch modes", "error": str(e)},
```

File: care/abdm/api/viewsets/healthid.py [hunk 3]
```diff
@@ -432,23 +444,29 @@ def add_care_context(self, request, *args, **kwargs):
...
+                    if "dob" in request.data
+                    else str(consultation.patient.abha_number.date_of_birth),
+                    "consultationId": consultation_id,
+                    # "authMode": "DIRECT",
+                    "purpose": "LINK",
+                }
+            )
+        except Exception as e:
+            return Response(
+                {"message": "Failed to add care context", "reason": str(e)},
```
Comment (sainak) [inline care/abdm/api/viewsets/healthid.py:467]:
```suggestion
                {"detail": "Failed to add care context", "error": str(e)},
```

File: care/abdm/api/viewsets/healthid.py [hunk 4]
```diff
@@ -470,9 +488,18 @@ def patient_sms_notify(self, request, *args, **kwargs):
...
+        try:
+            response = AbdmGateway().patient_sms_notify(
+                {
+                    "phone": patient.phone_number,
+                    "healthId": patient.abha_number.health_id,
+                }
+            )
+        except Exception as e:
+            return Response(
+                {"message": "Failed to send SMS", "reason": str(e)},
```
Comment (sainak) [inline care/abdm/api/viewsets/healthid.py:500]:
```suggestion
                {"detail": "Failed to send SMS", "error": str(e)},
```

File: care/abdm/api/viewsets/hip.py [hunk 5]
```diff
@@ -87,25 +87,35 @@ def share(self, request, *args, **kwargs):
...
+                                    f"{patient_data['yearOfBirth']}-{patient_data['monthOfBirth']}-{patient_data['dayOfBirth']}",
+                                    "%Y-%m-%d",
+                                )
+                            )[0:10],
+                        }
+                    )
+                except Exception:
+                    return Response(
+                        {
+                            "status": "FAILED",
```
Comment (sainak) [inline care/abdm/api/viewsets/hip.py:108]: todo for later: show proper message

**Review by sainak — APPROVED** [review]

**Comment by khavinshankar** [comment]
Test fails not related to the changes in this pr

**Review by gigincg — APPROVED** [review]

**Pull request merged**